### DISCUSSION TITLE
Can't understand why build is failing when kicked off manually via CLI only

### DISCUSSION THREAD

**Discussion by dgattey** [vercel/community#4137] [Bug Report]
### Page to Investigate

https://vercel.com/dgattey/dg/ECKJm6it3Wh8K4RCBsT8nEMcdMqg?filter=all

### Steps to Reproduce

I have a build that gets kicked off via a normal merge, and everything's working great with latest Next + all dependent packages and Turborepo. However, in my release flow in Github Actions, I kick off another build via `npx vercel`, and that one fails mysteriously on `pnpm install` with pretty minimal log information about what's happening. I can't get it to fail locally, is there any more insight the build logs provide for an employee of Vercel or anything common that would be occurring here?

Failing deployment: https://vercel.com/dgattey/dg/ECKJm6it3Wh8K4RCBsT8nEMcdMqg
Succeeding deployment: https://vercel.com/dgattey/dg/BLcM2S2PBqacvRGCSLDV1MzJ2Bhw
The Github action itself: https://github.com/dgattey/dg/blob/main/.github/workflows/release.yml

And yes, all the env variables specified in the action file are correct, I've verified/replaced the secrets

**Comment by TooTallNate** [comment]
When you deploy with CLI, the `.git` directory is not present in the build container, and it looks like a module that you are using, [`lefthook`](https://github.com/evilmartians/lefthook), doesn't want to install correctly without the `.git` directory being present.

One thing you could do is use `vc build` and `vc deploy --prebuilt` inside if you GH Action, instead of doing a "regular" CLI deployment. This guide walks you through that process, if you are interested: https://vercel.com/guides/how-can-i-use-github-actions-with-vercel

**Reply by dgattey** [reply]
Thanks for the quick reply! I'll try that out. I tried to disable the postinstall script with an inline check for `$CI` like Vercel's docs say should be there, but this didn't work:
```json
{
  "postinstall": "[ -z \"$CI\" ] && true || lefthook"
}
```

**Reply by dgattey** [reply]
@TooTallNate when I moved to this workflow file in my monorepo, it's now failing on `vercel build` with an obscure error and bad file paths:

This is the error with debug on:

```Vercel CLI 32.4.1
> [debug] [2023-10-08T00:23:35.507Z] user supplied known subcommand: "build"
> [debug] [2023-10-08T00:23:35.604Z] Found ".git/config" - detected "/home/runner/work/dg/dg" as repo root
> [debug] [2023-10-08T00:23:35.614Z] Loaded environment variables from "/home/runner/work/dg/dg/apps/web/.vercel/.env.production.local"
> [debug] [2023-10-08T00:23:35.624Z] Locating files /home/runner/work/dg/dg/apps/web/apps/web
> [debug] [2023-10-08T00:23:35.631Z] Ignoring invalid file /home/runner/work/dg/dg/apps/web/apps/web
> [debug] [2023-10-08T00:23:35.631Z] Locating files /home/runner/work/dg/dg/apps/web/apps/web [7ms]
> [debug] [2023-10-08T00:23:35.763Z] Imported Builder "@vercel/next" from "/home/runner/work/dg/dg/node_modules/.pnpm/@vercel+next@4.0.10/node_modules/@vercel/next"
> [debug] [2023-10-08T00:23:35.765Z] Setting env VERCEL_PROJECT_SETTINGS_BUILD_COMMAND to "turbo build"
> [debug] [2023-10-08T00:23:35.765Z] Setting env VERCEL_PROJECT_SETTINGS_INSTALL_COMMAND to "pnpm install"
> [debug] [2023-10-08T00:23:35.765Z] Setting env VERCEL_PROJECT_SETTINGS_NODE_VERSION to "18.x"
> [debug] [2023-10-08T00:23:35.765Z] Building entrypoint "package.json" with "@vercel/next"
Detected `pnpm-lock.yaml` version 6 generated by pnpm 8
Running "install" command: `pnpm install`...
Error: spawn sh ENOENT
Error: Process completed with exit code 1.
```

This is the workflow yml
```yml
# Tests a build and deploy to Vercel
name: Build and deploy to Vercel
on:
  - push

env:
  VERCEL_DEPLOY_TOKEN: ${{ secrets.VERCEL_DEPLOY_TOKEN }}
  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
  TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
  TURBO_TEAM: ${{ vars.TURBO_TEAM }}

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

jobs:
  release:
    timeout-minutes: 10
    name: Release
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - uses: pnpm/action-setup@v2
        with:
          run_install: false
      - uses: actions/setup-node@v3
        with:
          node-version-file: .node-version
          cache: 'pnpm'
      - run: pnpm install
      - uses: actions-ecosystem/action-get-latest-tag@v1
        name: Get latest version tag
        with:
          semver_only: true
        id: latest-version-tag

      - name: Pull Vercel Environment Information
        if: ${{ steps.latest-version-tag.outputs.tag != '' }}
        run: pnpm vercel pull --yes --cwd apps/web --environment=production --token=${{ env.VERCEL_DEPLOY_TOKEN }}
      - name: Build Project Artifacts with latest version
        if: ${{ steps.latest-version-tag.outputs.tag != '' }}
        run: |
          echo "Latest tag: ${LATEST_TAG}"
          pnpm vercel build --prod --cwd apps/web --token=${{ env.VERCEL_DEPLOY_TOKEN }} NEXT_PUBLIC_APP_VERSION="${LATEST_TAG}" -d
        env:
          LATEST_TAG: ${{ steps.latest-version-tag.outputs.tag }}
      - name: Deploy Project Artifacts to Vercel
        if: ${{ steps.latest-version-tag.outputs.tag != '' }}
        run: pnpm vercel deploy --prebuilt --prod --cwd apps/web --token=${{ env.VERCEL_DEPLOY_TOKEN }}
```

**Reply by dgattey** [reply]
Okay, looks like taking out `--cwd apps/web` did actually work, but I'm confused why that succeeded. Doesn't `vercel` need to know it should be run from a subfolder?
